testng-core/src/main/java/org/testng/internal/Yaml.java

```diff
@@ -347,4 +352,42 @@ public Object construct(Node node) {
       }
     }
   }
+
+  private static class XmlClassTypeDescriptor extends TypeDescription {
+
+    private final boolean loadClasses;
+
+    public XmlClassTypeDescriptor(boolean loadClasses) {
+      super(XmlClass.class);
+      this.loadClasses = loadClasses;
+    }
+
+    @Override
+    public Object newInstance(Node node) {
+      String className;
+
+      try {
+        java.lang.reflect.Constructor<?> c =
+            XmlClass.class.getDeclaredConstructor(String.class, boolean.class);
+        c.setAccessible(true);
+        if (node instanceof MappingNode) {
+          Node valueNode =
+              ((MappingNode) node)
+                  .getValue().stream()
+                      .filter(
+                          nodeTuple ->
+                              ((ScalarNode) nodeTuple.getKeyNode()).getValue().equals("name"))
+                      .findFirst()
+                      .orElseThrow(RuntimeException::new)
+                      .getValueNode();
+          className = ((ScalarNode) valueNode).getValue();
+        } else {
+          className = ((ScalarNode) node).getValue();
+        }
+        return c.newInstance(className, loadClasses);
```

Comment:
Can we please use `org.testng.internal.objects.InstanceCreator#newInstance(java.lang.reflect.Constructor<T>, java.lang.Object...)` to create the instance ?

Reply:
done